internal/security/anchore_client.go

```diff
@@ -317,6 +319,47 @@ func (a anchoreClient) AddRegistry(ctx context.Context, registry Registry) error
 	return nil
 }
 
+func (a anchoreClient) GetRegistry(ctx context.Context, registryName string) ([]anchore.RegistryConfiguration, error) {
+	a.logger.Info("getting anchore registry", map[string]interface{}{
+		"registryName": registryName,
+	})
+
+	opts := &anchore.GetRegistryOpts{}
+	registry, resp, err := a.getRestClient().RegistriesApi.GetRegistry(a.authorizedContext(ctx), registryName, opts)
+
+	a.logger.Debug("registry", map[string]interface{}{
+		"response":   resp.Status,
+		"registriey": registry,
+	})
+
+	if err != nil || (resp.StatusCode != http.StatusOK) {
+		return nil, errors.WrapIfWithDetails(err, "failed to get registry", registryName)
+	}
+
+	return registry, nil
+}
+
+func (a anchoreClient) UpdateRegistry(ctx context.Context, registry Registry) error {
+	fnCtx := map[string]interface{}{"registry": registry.Registry}
+	a.logger.Info("updating anchore registry", fnCtx)
+
+	// https://github.com/anchore/anchore-engine/issues/847
+	// using DeleteRegistry and AddRegistry instead of updateRegistry because UpdateRegistry doesn't work in anchore-engine
+	opts := &anchore.DeleteRegistryOpts{}
+	resp, err := a.getRestClient().RegistriesApi.DeleteRegistry(a.authorizedContext(ctx), registry.Registry, opts)
+	if err != nil || (resp.StatusCode != http.StatusOK) {
+		return errors.WrapIfWithDetails(err, "failed to delete anchore registry", fnCtx)
+	}
+
+	err = a.AddRegistry(ctx, registry)
+	if err != nil {
+		return err
+	}
```

Comment:
We don't loose any state information or functionality by this workaround right?
(I suppose we don't, this is just a "for the record I feel safer to ask explicitly" question.)

Reply:
We don't lose. If a registry isn't changed, we will not do anything with it. Currently, this is the only way we can update a registry in anchore-engine.